File: src/app/models/course/course-response.ts
import {LessonResponse} from "../lesson/lesson-response";

export interface CourseResponse {
  id: number;
  title: string;
  description: string;
  createdAt: string; // You may need to adjust the type based on your date format
  updatedAt: string; // You may need to adjust the type based on your date format
  topicId: number;
  topicTitle: string;
  imageFile: string;
  startDate: string; // You may need to adjust the type based on your date format
  endDate: string; // You may need to adjust the type based on your date format
  free: boolean;
 // price: number;
  timeCommitment: string;
  publisherName: string;
  lessons: LessonResponse[]; // Assuming LessonResponse is another interface

}

File: src/main/java/com/pfe/elearning/course/dto/CourseResponse.java
package com.pfe.elearning.course.dto;

import com.pfe.elearning.lesson.dto.LessonResponse;
import com.pfe.elearning.topic.dto.TopicResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;


@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor

public class CourseResponse {
    private Integer id;
    private String title;
    private String description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Integer topicId;
    private String topicTitle;
    private String imageFile;
    private LocalDate startDate;
    private LocalDate endDate;
    private boolean free;
    private Float price;
    private String timeCommitment;
    private String publisherName;
    //private TopicResponse topic;
    private List<LessonResponse> lessons;


    private int courseCount;



}